tests/unit/test_routes.py

```diff
@@ -1,6 +1,9 @@
 import pytest, random
 from tests import conftest
 import time
+import os, io
+import yaml
+import re
```

Comment:
I assume this is still a work in progress and that we're only going to keep imports that we actually use at the end of it all?

Reply:
yes i still have not evaluated the yaml file. I wanted to see what you though about evaluating the headers? So with evaluating the yaml file I am trying to find a way to mock the behavior or capture the downloaded file without pulling from an actual path unless its sent to a tempdir

Reply:
See my comment above on evaluating the headers. I'm indifferent. Feel free to leave it in if you like, but it doesn't really verify that the endpoint is working correctly so until we have the yaml.load() this still can't be merged.